### ISSUE TITLE
Contribute to HALO Database

### ISSUE THREAD

**Issue by joerg-halo** [halo-db/storymap#44]
As a user of the database I have ideas for the improvements of the HALO Database from time to time. When I have capacity to implement these ideas, I would like to do so in order ease my work with HALO data and to help the HALO community with these contributions

**Comment by bjoernbroetz** [comment]
Feedback button! Good!

**Comment by d70-t** [comment]
This is a great idea. The entire HALO-DB codebase should be open sourced such that users can look into why things behave as they to and could directly improve the database by the creation of pull requests.

Furthermore, if the concept of the database turns out to be a good one, the other institutions might be interested in using it as well. This might be beneficial for interoperability (e.g. #37)

**Comment by joerg-halo** [comment]
This use case was raised in the survey about the HALO database. The wish was, that the source code of the website is open source.

**Comment by d70-t** [comment]
Actually, this is even a hard requirement as per the current Halo Data Policy, quoting §2 (2 & 4):

> (2) Data repository: The repository for the Data of a mission is the HALO Database, [...]

> (4) Data format: The Data must be provided in a form that ensures accessibility without proprietary tools/software. [...]

So the data is to be provided via the HALO Database and it must be provided without proprietary tools/software, hence all software required to run the HALO Database must not be proprietary and thus open.

**Comment by joernu76** [comment]
> Actually, this is even a hard requirement as per the current Halo Data Policy, quoting §2 (2 & 4):
> 
> > (2) Data repository: The repository for the Data of a mission is the HALO Database, [...]
> 
> > (4) Data format: The Data must be provided in a form that ensures accessibility without proprietary tools/software. [...]
> 
> So the data is to be provided via the HALO Database and it must be provided without proprietary tools/software, hence all software required to run the HALO Database must not be proprietary and thus open.

I disagree with the interpretation of this requirement. The *usage* of the database must be possible without proprietary tools, which is given by offering a website that is accessible with any browser. 
but I deem it also beneficial to use as much as possible/only open source tools, if reasonable. Programming our own framework is likely out-of-scope, however...?

**Comment by joernu76** [comment]
**Consensus of AG HALO Database:**

Priorization: Nice to have

We deem it useful to have all code/configuration/documentation in (a set of) github (or similar) write-protected repositories.
Feedback (e.g. merge requests) handling and governance may be discussed at a later point in time.
